src/components/Api.js

```diff
@@ -0,0 +1,73 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this.server = baseUrl;
+    this.headers = headers;
+  }
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  getInitialCards() {
+    return fetch(`${this.server}/cards`, {
+      headers: this.headers,
+    }).then(this._checkResponse);
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.

Reply:
> The code breaks when I try to do this.

This is how it should be used 


```js
  _request(url, options) {
    return fetch(url, options).then(this._checkResponse);
  }

  getInitialCards() {
    return this._request(`${this.server}/cards`, {
      headers: this.headers,
    });
  }

  getUserInfo() {
    return this._request(`${this.server}/users/me`, {
      headers: this.headers,
    });
  }

```